LearningMissionSimulation/AttributeGenerator.cs

```diff
@@ -298,6 +298,65 @@ public static string GetCoverLetter()
             //not implemented yet!
             return "";
         }
+        public static string GetUnitName()
+        {
+            string unitName = GetFirstName();
+
+            return unitName;
+        }
+        public static string GetUnitDescprition(UnitType unitType)
+        {            
+            int i = SimulationConstants.random.Next(5, 25);
+            int j = 0;
+            string unitTypeDiscription = "";
+            switch (unitType)
+            {
+                case UnitType.Unspecified:
+                    unitTypeDiscription = "not have a discription";
```

Comment:
it is not a good way of handling the unspecified type. Use "Unspecified" in this case.
And you don't need a switch statement. You can handle it with a single line of code:
var type = UnitType.Module + "1234"; // the result is "Module1234"